graylog2-server/src/main/java/org/graylog/scheduler/DBJobTriggerService.java

```diff
@@ -401,50 +403,48 @@ public boolean releaseTrigger(JobTriggerDto trigger, JobTriggerUpdate triggerUpd
+                eq(FIELD_LOCK_OWNER, nodeId),
+                idEq(getId(trigger)),
                 // Only release running triggers. The trigger might have been paused while the trigger was running
                 // so we don't want to set it to RUNNABLE again.
                 // TODO: This is an issue. If a user set it to PAUSED, we will not unlock it. Figure something out.
                 //       Maybe a manual trigger pause will set "nextStatus" if the trigger is currently running?
                 //       That next status would need to be set on release.
-                DBQuery.is(FIELD_STATUS, JobTriggerStatus.RUNNING)
+                eq(FIELD_STATUS, JobTriggerStatus.RUNNING)
         );
-        final DBUpdate.Builder update = DBUpdate.unset(FIELD_LOCK_OWNER);
+
+        final List<Bson> updates = new ArrayList<>();
+        updates.add(unset(FIELD_LOCK_OWNER));
 
         if (triggerUpdate.concurrencyReschedule()) {
-            update.inc(FIELD_CONCURRENCY_RESCHEDULE_COUNT);
+            updates.add(inc(FIELD_CONCURRENCY_RESCHEDULE_COUNT, 1));
         } else {
-            update.set(FIELD_CONCURRENCY_RESCHEDULE_COUNT, 0);
+            updates.add(set(FIELD_CONCURRENCY_RESCHEDULE_COUNT, 0));
         }
 
         // An empty next time indicates that this trigger should not be fired anymore. (e.g. for "once" schedules)
         if (triggerUpdate.nextTime().isPresent()) {
             if (triggerUpdate.status().isPresent()) {
-                update.set(FIELD_STATUS, triggerUpdate.status().get());
+                updates.add(set(FIELD_STATUS, triggerUpdate.status().get()));
             } else {
-                update.set(FIELD_STATUS, JobTriggerStatus.RUNNABLE);
+                updates.add(set(FIELD_STATUS, JobTriggerStatus.RUNNABLE));
             }
-            update.set(FIELD_NEXT_TIME, triggerUpdate.nextTime().get());
+            updates.add(set(FIELD_NEXT_TIME, triggerUpdate.nextTime().get()));
         } else {
-            update.set(FIELD_STATUS, triggerUpdate.status().orElse(JobTriggerStatus.COMPLETE));
+            updates.add(set(FIELD_STATUS, triggerUpdate.status().orElse(JobTriggerStatus.COMPLETE)));
         }
 
         if (triggerUpdate.data().isPresent()) {
-            update.set(FIELD_DATA, triggerUpdate.data());
+            updates.add(set(FIELD_DATA, triggerUpdate.data()));
         }
         trigger.triggeredAt().ifPresent(triggeredAt -> {
             var duration = new org.joda.time.Duration(triggeredAt, clock.nowUTC());
-            update.set(FIELD_LAST_EXECUTION_DURATION, Optional.of(duration.getMillis()));
+            updates.add(set(FIELD_LAST_EXECUTION_DURATION, Optional.of(duration.getMillis())));
         });
 
-        final int changedDocs = db.update(query, update).getN();
-        if (changedDocs > 1) {
-            throw new IllegalStateException("Expected to release only one trigger (id=" + trigger.id() + ") but database query modified " + changedDocs);
-        }
```

Comment:
Could you let me know if you deliberately removed the check?

Reply:
Yes, that was deliberately.

We are doing a single-document update. Implicitly, by not setting the `multi` flag to `true` in `db.update(...)` in the old code, and explicitly by calling `updateOne(...)` in the new code.

Therefore, I don't think `changedDocs` could have ever been > 1, or am I missing something?